servers/src/test/java/tachyon/master/block/BlockMasterTest.java

```diff
@@ -220,6 +227,51 @@ public void unknownHeartbeatTest() throws Exception {
     Assert.assertEquals(new Command(CommandType.Register, ImmutableList.<Long>of()), heartBeat);
   }
 
+  @Test
+  public void detectLostWorkerTest() throws Exception {
+    HeartbeatScheduler.await(HeartbeatContext.MASTER_LOST_WORKER_DETECTION, 5, TimeUnit.SECONDS);
+
+    // Get a new worker id.
+    long workerId = mMaster.getWorkerId(new NetAddress("localhost", 80, 81));
+    MasterWorkerInfo workerInfo = mPrivateAccess.getWorkerById(workerId);
+    Assert.assertNotNull(workerInfo);
+
+    // Run the lost worker detector. No workers will be considered lost.
+    HeartbeatScheduler.schedule(HeartbeatContext.MASTER_LOST_WORKER_DETECTION);
+    Assert.assertTrue(HeartbeatScheduler.await(HeartbeatContext.MASTER_LOST_WORKER_DETECTION, 1,
+        TimeUnit.SECONDS));
+
+    // No lost workers should be considered lost.
+    Assert.assertEquals(0, mMaster.getLostWorkersInfo().size());
+    Assert.assertNotNull(mPrivateAccess.getWorkerById(workerId));
+
+    // Set the last updated time for the worker to be old, so it is considered lost.
+    workerInfo.updateLastUpdatedTimeMs(System.currentTimeMillis()
+        - 2 * MasterContext.getConf().getInt(Constants.MASTER_WORKER_TIMEOUT_MS));
+
+    // Run the lost worker detector. The worker should be considered lost.
+    HeartbeatScheduler.schedule(HeartbeatContext.MASTER_LOST_WORKER_DETECTION);
+    Assert.assertTrue(HeartbeatScheduler.await(HeartbeatContext.MASTER_LOST_WORKER_DETECTION, 1,
+        TimeUnit.SECONDS));
+
+    // There should be a lost worker.
+    Assert.assertEquals(1, mMaster.getLostWorkersInfo().size());
+    Assert.assertNull(mPrivateAccess.getWorkerById(workerId));
+
+    // Get the worker id again, simulating the lost worker re-registering.
+    workerId = mMaster.getWorkerId(new NetAddress("localhost", 80, 81));
+    Assert.assertNotNull(mPrivateAccess.getWorkerById(workerId));
+
+    // Run the lost worker detector. No workers will be considered lost.
```

Comment:
The `No workers ...` part of the comment is repeated below and seems unnecessary here.